common/src/api/external/mod.rs

```diff
@@ -2126,6 +2148,376 @@ impl std::fmt::Display for Digest {
+    /// The id of this LLDP service instance.
+    pub id: Uuid,
+
+    /// The link-layer discovery protocol configuration for this service.
+    pub lldp_config_id: Option<Uuid>,
+
+    /// Whether or not the LLDP service is enabled.
+    pub enabled: bool,
+}
+
+/// A link layer discovery protocol (LLDP) base configuration.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct LldpConfig {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+
+    /// The LLDP chassis identifier TLV.
+    pub chassis_id: String,
+
+    /// THE LLDP system name TLV.
+    pub system_name: String,
+
+    /// THE LLDP system description TLV.
+    pub system_description: String,
+
+    /// THE LLDP management IP TLV.
+    pub management_ip: IpNet,
+}
+
+/// Describes the kind of an switch interface.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum SwitchInterfaceKind {
+    /// Primary interfaces are associated with physical links. There is exactly
+    /// one primary interface per physical link.
+    Primary,
+
+    /// VLAN interfaces allow physical interfaces to be multiplexed onto
+    /// multiple logical links, each distinguished by a 12-bit 802.1Q Ethernet
+    /// tag.
+    Vlan,
+
+    /// Loopback interfaces are anchors for IP addresses that are not specific
+    /// to any particular port.
+    Loopback,
+}
+
+/// A switch port interface configuration for a port settings object.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct SwitchInterfaceConfig {
+    /// The port settings object this switch interface configuration belongs to.
+    pub port_settings_id: Uuid,
+
+    /// A unique identifier for this switch interface.
+    pub id: Uuid,
+
+    /// The name of this switch interface.
+    pub interface_name: String,
+
+    /// Whether or not IPv6 is enabled on this interface.
+    pub v6_enable: bool,
+
+    /// The switch interface kind.
+    pub kind: SwitchInterfaceKind,
+}
+
+/// A switch port VLAN interface configuration for a port settings object.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct SwitchVlanInterfaceConfig {
+    /// The switch interface configuration this VLAN interface configuration
+    /// belongs to.
+    pub interface_config_id: Uuid,
+
+    /// The virtual network id (VID) that distinguishes this interface and is
+    /// used for producing and consuming 802.1Q Ethernet tags. This field has a
+    /// maximum value of 4095 as 802.1Q tags are twelve bits.
+    pub vid: u16,
+}
+
+/// A route configuration for a port settings object.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq)]
+pub struct SwitchPortRouteConfig {
+    /// The port settings object this route configuration belongs to.
+    pub port_settings_id: Uuid,
+
+    /// The interface name this route configuration is assigned to.
+    pub interface_name: String,
```

Comment:
As with previous notes, it may be worth just making this a `Name` type for consistency even if the limitation isn't necessary.

Reply:
https://github.com/oxidecomputer/omicron/issues/3050